compiler-core/src/language_server/server.rs

```diff
@@ -181,12 +205,57 @@ where
                 self.watched_files_changed(params)
             }
 
+            "workspace/didChangeConfiguration" => {
+                self.request_configuration();
+                Feedback::default()
+            }
+
             _ => return,
         };
 
         self.publish_feedback(feedback);
     }
 
+    fn handle_response(&mut self, handler: ResponseHandler, result: Json) {
+        match handler {
+            ResponseHandler::UpdateConfiguration => {
+                match serde_json::from_value::<Vec<Configuration>>(result) {
+                    Ok(result) => {
+                        if let Some(config) = result.into_iter().next() {
+                            self.config = VersionedConfig {
+                                version: self.config.version + 1,
```

Comment:
What's the version thing here?

Reply:
Its used to quickly check if the config cached in the engine is out of date. Its checked in the `respond_with_engine` function to lazily update the config if the versions don't match. An alternative might be to eagerly loop through the engines and update the cached configs, or I suppose share the config with an Arc RwLock or similar. I don't have a particular preference here

Reply:
Oh cool! Could we use the RwLock approach? That sounds nice to me.